tests/pytest/test_multi.py

```diff
@@ -978,139 +978,316 @@ def testErrorMessage(env):
     r.expect('JSON.ARRLEN', 'doc1', '$.nowhere', '"abc"').raiseError().contains("does not exist")
     r.expect('JSON.ARRLEN', 'doc_none', '$.string', '"abc"').raiseError().contains("doesn't exist")
+    r.expect('JSON.ARRLEN', 'hash_key', '$.string', '"abc"').raiseError().contains("wrong Redis type")
 
-    r.assertEqual(r.execute_command('JSON.ARRLEN', 'doc1', '.string', '"abc"'), None)
+    r.expect('JSON.ARRLEN', 'doc1', '.string', '"abc"').raiseError().contains("not an array")
     r.expect('JSON.ARRLEN', 'doc1', '.nowhere', '"abc"').raiseError().contains("does not exist")
     r.assertEqual(r.execute_command('JSON.ARRLEN', 'doc_none', '.string', '"abc"'), None)
+    """ Legacy 1.0.8:
+    json.set doc1 .  '{"a":[0, 1, 2, 3, 4, 5], "b":{"x":100}}'
+    OK
+    json.arrlen doc1 .b
+    (error) ERR wrong type of path value - expected array but found object
+    json.arrlen doc1 .zzz
+    (error) ERR key 'zzz' does not exist at level 0 in path
+    json.arrlen doc1zz .zzz
+    (nil)
+    """
 
-    # ARRTRIM
+# ARRTRIM
     r.assertEqual(r.execute_command('JSON.ARRTRIM', 'doc1', '$.string', 0, 1), [None])
     r.expect('JSON.ARRTRIM', 'doc1', '$.nowhere', 0, 1).raiseError().contains("does not exist")
     r.expect('JSON.ARRTRIM', 'doc_none', '$.string', 0, 1).raiseError().contains("doesn't exist")
+    r.expect('JSON.ARRTRIM', 'hash_key', '$.string', 0, 1).raiseError().contains("wrong Redis type")
 
     r.expect('JSON.ARRTRIM', 'doc1', '.string', 0, 1).raiseError().contains("not an array")
     r.expect('JSON.ARRTRIM', 'doc1', '.nowhere', 0, 1).raiseError().contains("not an array")
     r.expect('JSON.ARRTRIM', 'doc_none', '.string', 0, 1).raiseError().contains("doesn't exist")
+    """ Legacy 1.0.8:
+    json.set doc1 .  '{"a":[0, 1, 2, 3, 4, 5], "b":{"x":100}}'
+    OK
+    json.arrtrim doc1 .b 0 3
+    (error) ERR wrong type of path value - expected array but found object
+    json.arrtrim doc1 .bzzz 0 3
+    (error) ERR key 'bzzz' does not exist at level 0 in path
+    json.arrtrim doc1zzz .b 0 3
+    (error) WRONGTYPE Operation against a key holding the wrong kind of value
+    """
 
     # OBJKEYS
     r.assertEqual(r.execute_command('JSON.OBJKEYS', 'doc1', '$.string'), [None])
     r.expect('JSON.OBJKEYS', 'doc1', '$.nowhere').raiseError().contains("does not exist")
-    r.assertEqual(r.execute_command('JSON.OBJKEYS', 'doc_none', '$.string'), None)
+    r.expect('JSON.OBJKEYS', 'doc_none', '$.string').raiseError().contains("doesn't exist")
+    r.expect('JSON.OBJKEYS', 'hash_key', '$.string').raiseError().contains("wrong Redis type")
 
-    r.expect('JSON.OBJKEYS', 'doc1', '.string').raiseError().contains("expected object but found string")
-    r.expect('JSON.OBJKEYS', 'doc1', '.nowhere').raiseError().contains("does not exist")
+    r.expect('JSON.OBJKEYS', 'doc1', '.string').raiseError().contains("not an object")
+    r.assertEqual(r.execute_command('JSON.OBJKEYS', 'doc1', '.nowhere'), None)
```

Comment:
```suggestion
    r.expect('JSON.OBJKEYS', 'doc1', '.nowhere').equal(None)
```

Reply:
Keeping `execute_command` to handle cluster connection